### DISCUSSION TITLE
Retry mechanism documentation case

### DISCUSSION THREAD

**Discussion by image-gif** [midwayjs/midway#3654] [General / 闲聊]
In the document case, some minor problems were found;

![image](https://github.com/midwayjs/midway/assets/74607022/ad42eaf6-27f4-4bb3-a266-ddb986f0391f)

It is easy to cause ambiguity here. It is recommended to use other function names instead and improve the cases.
like this:
```typescript
import { retryWithAsync } from '@midwayjs/core';

async function foo(val: number) {

  // TODO: asynchronous operations

}

async function invoke() {
  // 默认调用，加上重试两次，最多执行三次
  const invokeNew = retryWithAsync(foo, 2);

  try {
    return await invokeNew(1);
  } catch(err) {

    // err
  }
}
```
The above only represents my personal opinion. If there are any errors, please let me know. Thank you.

**Comment by czy88840616** [comment]
LGTM

**Comment by czy88840616** [comment]
has been updated
